src/Microsoft.Identity.Client/ApiConfig/Executors/ClientApplicationBaseExecutor.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Exceptions;
+using Microsoft.Identity.Client.Internal.Requests;
+
+namespace Microsoft.Identity.Client.ApiConfig.Executors
+{
+    internal class ClientApplicationBaseExecutor : AbstractExecutor, IClientApplicationBaseExecutor
+    {
+        private readonly ClientApplicationBase _clientApplicationBase;
```

Comment:
You could make this protected in the base class.

Reply:
I can't do that for the other two, so I decided to make the implementations at the leaf-level consistent by repeating this particular reference.